src/pydata_sphinx_theme/theme/pydata_sphinx_theme/components/edit-this-page.html

```diff
@@ -1,8 +1,16 @@
 {% if sourcename is defined and theme_use_edit_page_button==true and page_source_suffix %}
   {% set src = sourcename.split('.') %}
   <div class="tocsection editthispage">
-    <a href="{{ get_edit_url() }}">
-      <i class="fa-solid fa-pencil"></i> {{ _("Edit this page") }}
+    <a href="{{ get_edit_provider_and_url()[1] }}">
+      <i class="fa-solid fa-pencil"></i>
+      {% set provider = get_edit_provider_and_url()[0] %}
+      {% block edit_this_page_text %}
+        {% if provider %}
+          {% trans provider=provider %}Edit on {{ provider }}{% endtrans %}
```

Comment:
I think that here you want to use the in-built Sphinx function to define translations, which is the (unhelpfully named) `_` function. Then you'll want to only wrap the things that must be translated because it treats the full wrapped block as a "unique" translation string. E.g. if you do `_("Edit on GitHub")` and `_("Edit on GitLab")` it treats these as two totally different strings. I think instead you'd want `_("Edit on") GitHub` in the template. Does that make sense?

Reply:
ah nice - you clearly know more about this than I do. I was just trying to figure out how to avoid needing to define unique translation files for "GitHub", "GitLab", etc :-)